include/h2o/mruby_.h

```diff
@@ -177,10 +182,18 @@ mrb_value h2o_mruby_http_join_response_callback(h2o_mruby_context_t *ctx, mrb_va
                                                 int *next_action);
 mrb_value h2o_mruby_http_fetch_chunk_callback(h2o_mruby_context_t *ctx, mrb_value receiver, mrb_value input,
                                               int *next_action);
+mrb_value h2o_mruby_channel_shift_callback(h2o_mruby_context_t *ctx, mrb_value receiver, mrb_value input,
+                                              int *next_action);
```

Comment:
How about retaining the queue in mruby side, while restricting use of the C code to when signalling between fibers is necessary?

I believe that doing so would simplify the code and will give us more flexibility in extending the object (for example, it would be easier to add a `peek` method). 

```ruby
class Channel
  def initialize()
    @queue = []
  end
  def push(o)
    @queue << o
    self._notify
  end
  def shift
    while true
      if !@queue.empty?
        return @queue.shift
      end
      self._wait
    end
  end
  // _notify and _wait are C methods that are called via fiber switching
end
```